src/onegov/fsi/upgrade.py

```diff
@@ -172,3 +174,125 @@ def add_active_property_to_attendees(context):
             Column('active', Boolean, nullable=False, default=True),
             default=lambda x: True
         )
+
+
+p2 = re.compile(r'(.*), (.*)Postadresse: (.*), (.*)')
+p3 = re.compile(r'(.*), (Postfach), (.*)')
+p4 = re.compile(r'(.*), (.*), (.*)')
+p1 = re.compile(r'(.*), (.*)')
+p5 = re.compile(r'([A-Za-z ]*) ?(\d+[a-z]?)?')  # street name and optional
+# building number
+
+
+def parse_and_split_address_field(address):
+    """
+    Parsing the `address` field to split into location address and code/city
+    as well as postal address and code/city.
+
+    :param address:str
+    :return: tuple: (location_address, location_code_city,
+                     postal_address, postal_code_city)
+    """
+    location_addr = ''
+    location_pcc = ''
+    postal_addr = ''
+    postal_pcc = ''
+
+    # sanitize address
+    if ';' in address:
+        address = address.replace('; ', '')
+        address = address.replace(';', '')
+
+    if not address:
+        return (location_addr, location_pcc, postal_addr, postal_pcc)
+
+    if m := p2.match(address):
+        location_addr = m.group(1)
+        location_pcc = m.group(2)
+        postal_addr = m.group(3)
+        postal_pcc = m.group(4)
+        return (location_addr, location_pcc, postal_addr, postal_pcc)
+
+    if m := p3.match(address):
+        postal_addr = m.group(1) + '\n' + m.group(2)
+        postal_pcc = m.group(3)
+        return (location_addr, location_pcc, postal_addr, postal_pcc)
+
+    if m := p4.match(address):
+        postal_addr = m.group(1) + '\n' + m.group(2)
+        postal_pcc = m.group(3)
+        return (location_addr, location_pcc, postal_addr, postal_pcc)
+
+    if m := p1.match(address):
+        postal_addr = m.group(1)
+        postal_pcc = m.group(2)
+        return (location_addr, location_pcc, postal_addr, postal_pcc)
+
+    if m := p5.match(address):
+        postal_addr = m.group(1)
+        if m.group(2):
+            postal_addr += f'{m.group(2)}'
+        return (location_addr, location_pcc, postal_addr, postal_pcc)
+
+    # default
+    print(f'*** parse_and_split_address_field: no match found for address:'
+          f' {address}')
+    return location_addr, location_pcc, postal_addr, postal_pcc
+
+
+@upgrade_task('ogc-966 extend agency and person tables with more fields')
+def extend_agency_and_person_with_more_fields(context):
+    ################################################################
```

Comment:
We don't use such marking comments anywhere...

Reply:
removed